File: frontend/src/app/models/recurring-expense.ts
import { Category, PaymentMethod } from './expense';

export type RecurringFrequency = 'Daily' | 'Weekly' | 'Monthly' | 'Yearly';

export interface RecurringExpense {
  id: number;
  title: string;
  amount: number;
  category: Category;
  paymentMethod: PaymentMethod | null;
  frequency: RecurringFrequency;
  startDate: string;
  endDate: string | null;
  lastExecutedDate: string | null;
}

File: backend/Models/RecurringExpense.cs
namespace ExpenseTracker.API.Models;

public enum RecurringFrequency { Daily, Weekly, Monthly, Yearly }

public class RecurringExpense
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public Category Category { get; set; }
    public PaymentMethod? PaymentMethod { get; set; }
    public RecurringFrequency Frequency { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public DateOnly? LastExecutedDate { get; set; }
}
